### PR TITLE
Recommendation 3.6 - usage of blank lines

### LINKED ISSUE #150: Recommendation 3.6: Reduce unnecessary blank lines and keep the code compact
By reducing unnecessary blank lines, you can display more code on one screen and that makes the code more readable.

- Blank lines should separate content based on it's relevance: blank lines should be placed between groups of fields, constructors, methods, nested classes, init blocks, objects (see [rule 3.2](#r3.2))
- Do not use more than one single line inside methods, type definitions, initialization expressions
- Generally do not use more than two consecutive blank lines in a row
- Do not put newlines in the beginning or at the end of code blocks with curly braces:

```kotlin
fun baz() {
        
    doSomething();  // No need to add blank lines at the beginning and at the end of the code block
    // ...

}
```

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Added new rule
* Added tests
* Updated rules-config.json, available-rules.md

## Which rule and warnings did you add?

This pull request closes #150 
This rule prohibits usage of two or more consecutive blank lines. It also prohibits blank line in the beginning or at the end of code block.

## Actions checklist
* [x] Implemented Rule, added Warnings
* [x] Added tests on checks
* [x] Added tests on fixers
* [x] Updated rules-config.json
* [x] Updated available-rules.md

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import com.pinterest.ktlint.core.ast.ElementType.RBRACE
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.TOO_MANY_BLANK_LINES
+import org.cqfn.diktat.ruleset.rules.getDiktatConfigRules
+import org.cqfn.diktat.ruleset.utils.leaveExactlyNumNewLines
+import org.cqfn.diktat.ruleset.utils.leaveOnlyOneNewLine
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+class BlankLinesRule : Rule("blank-lines") {
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt:17]: please do not forget to add description to rules

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt:23]: this visitor should check and fix that code blocks are separated by new lines, please create a feature for this. 
We don't want to have such code:
```
package 
import
fun foo() {
}
fun foo1() {
}
```

we need to fix such code

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt:23]: if you think that should be in a separate rule - let's add such rule to a code style

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt:23]: I believe it's [ClassLikeStructuresOrderRule](https://github.com/cqfn/diKTat/blob/master/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/ClassLikeStructuresOrderRule.kt) that already does this - it rearranges code blocks and separates them with newlines.

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/BlankLinesRule.kt:23]: It's [Rule 3.2](https://github.com/cqfn/diKTat/blob/master/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/ClassLikeStructuresOrderRule.kt) of code style.

**Pull request merged**